bin/detect

```diff
@@ -13,7 +13,38 @@ get_client() {
   fi
 }
 
+write_build_plan() {
+  local plan=${1:?}
+  local appDir="$(pwd)"
+  local bpDir="$(cd $(dirname $0)/..; pwd)" # absolute path
+  source "${bpDir}/lib/jvm.sh"
+
+  local jdkVersion="$(get_jdk_version "${appDir}")"
+  local fullJdkVersion="$(get_full_jdk_version "${jdkVersion}")"
+
+  cat <<TOML > $plan
+[[provides]]
+name = "jdk"
+
+[[requires]]
+name = "jdk"
+version = "$fullJdkVersion"
+[[requires.metadata]]
+launch = false
```

Comment:
what's the point of the metadata here?

Reply:
I've seen this used in the CF buildpacks, so I'm mostly mimicking. In the future (see the `#TODO`) the `bin/build` could set the layer visibility based on this value.

Reply:
this launch metadata on the build plan or the layer? It just seems leaky.